model/stats/db.go

```diff
@@ -228,16 +228,26 @@ func getHourlyTestStatsPipeline(projectId string, requester string, start time.T
 					{"$eq": Array{testResultTaskIdKeyRef, "$$task_id"}},
 					{"$eq": Array{testResultExecutionRef, "$$execution"}}}}}},
 				{"$project": bson.M{
-					testresult.IDKey:        0,
-					dbTestStatsLastIDKey:    "$" + testresult.IDKey,
-					testresult.TestFileKey:  1,
-					testresult.StatusKey:    1,
-					testresult.StartTimeKey: 1,
-					testresult.EndTimeKey:   1}}},
+					testresult.IDKey:              0,
+					dbTestStatsLastIDKey:          "$" + testresult.IDKey,
+					testresult.TestFileKey:        1,
+					testresult.DisplayTestNameKey: 1,
+					testresult.StatusKey:          1,
+					testresult.StartTimeKey:       1,
+					testresult.EndTimeKey:         1}}},
 			"as": "testresults"}},
 		{"$unwind": "$testresults"},
 		{"$project": bson.M{
-			dbTestStatsIdTestFileKey: "$testresults." + testresult.TestFileKey,
+			// Use the display test name if there is one.
+			dbTestStatsIdTestFileKey: bson.M{
+				"$cond": bson.M{
+					"if": bson.M{
+						"$ne": []string{bsonutil.GetDottedKeyName("$testresults", testresult.DisplayTestNameKey), ""},
```

Comment:
I'm not that familiar with this syntax since usually, it's expressed as `"field": {"$ne": <value>}`. Is it okay to use a string slice instead of an interface slice (i.e. `Array`) like the other places do in the test stats code?

Reply:
I thought it works this way because it's [the aggregation version](https://docs.mongodb.com/manual/reference/operator/aggregation/ne/)

Reply:
This should work in the same way that string slices are marshaled into valid BSON arrays when in our schema structs. I am double checking in staging.